core/domain/learner_playlist_services_test.py

```diff
@@ -100,162 +101,148 @@ def _get_all_learner_playlist_collection_ids(self, user_id):
             learner_playlist_model.collection_ids if
             learner_playlist_model else [])
 
-    def test_mark_exploration_to_be_played_later(self):
+    def test_adding_subscribed_exp_to_playlist_does_nothing_without_pos(self):
+        # Subscribe to exploration.
+        subscription_services.subscribe_to_exploration(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Now if we try to add the same exploration, it shouldn't be added.
+        learner_progress_services.add_exp_to_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
         self.assertEqual(
             self._get_all_learner_playlist_exp_ids(self.user_id), [])
 
-        # Add an exploration to the learner playlist.
+    def test_adding_existing_exp_to_playlist_does_not_add_again_with_pos(self):
```

Comment:
Generally speaking I think the "with/without pos" stuff is confusing, since the behavior is typically the same whether pos is specified or not.

So I suggest that you combine both the "with/without pos" stuff into the same test, unless specifying a pos actually changes the outcome. You could have two "actions" in the test, one without pos and one with pos (make sure to do a separate verification after each).

The shortening of the test names might also allow you to unabbreviate exp --> exploration and col --> collection.

Reply:
Ahhh I see. I'll combine them to be test_adding_existing_exploration_to_playlist_does_not_add_again(self):

Reply:
Sounds good, note that this comment applies elsewhere too.

Reply:
Okay.